### PR TITLE
Updated most of the smart contracts scripts to typescript. Updated th…

### PR DESCRIPTION
**Pull request by Brando753**
Integrate the blocklist deploy into the future/peggy2 branch

### CODE REVIEW

File: smart-contracts/tasks/blocklist/ofacParser.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const response = await axios.get(finalUrl).catch((e) => {
+    throw e;
+  });
+
+  const addresses = extractAddresses(response.data);
+
+  return addresses;
+}
+
+export function extractAddresses(rawFileContents: string) {
```
Comment (jzvikart) [inline smart-contracts/tasks/blocklist/ofacParser.ts:28]: Wasn't there also a formatted version of OFAC blocklist file? I thought somebody mentioned it a while ago in a meeting...

Comment (Brando753) [reply smart-contracts/tasks/blocklist/ofacParser.ts:28]: We had a script that would write these results to a file, but I deleted it as I didn't see it used for anything nor could I understand the use case.

Comment (jzvikart) [reply smart-contracts/tasks/blocklist/ofacParser.ts:28]: No, I think I remember seeing the actual JSON/XML file or something... I'm not 100% sure, but try asking James

Comment (Brando753) [reply smart-contracts/tasks/blocklist/ofacParser.ts:28]: You sure it wasn't a file generated by https://github.com/Sifchain/sifnode/blob/716e55c0f28ca3ae142706ee031d3bdf3aa1b8b8/smart-contracts/scripts/download_ofac_blocklist.js That would make a JSON file.

**Comment by jzvikart** [comment]
How much do we really need Hardhat here, and for what? It would be nice to decouple from hardhat as much as possible, especially since we're thinking about using geth as the main implementation.

**Comment by Brando753** [comment]
> How much do we really need Hardhat here, and for what? It would be nice to decouple from hardhat as much as possible, especially since we're thinking about using geth as the main implementation.

Hardhat lets us trivially test locally and compiles our smart contracts, and it provides typing through the typechain plugin, which is super useful when scripting. Just because the load tests will use geth rather than a hardhat node does not mean we need to decouple hardhat from our tests/scripts. Hardhat takes any network RPC in its config. Just as I can point hardhat to Ethereum mainnet I can point it to your local geth instance. I see no reason to decouple it unless hardhat itself is causing issues and not just the hardhat node instance.

File: smart-contracts/tasks/blocklist/ofacParser.ts (hunk 1 above)
Comment (banshee) [reply smart-contracts/tasks/blocklist/ofacParser.ts:28]: OFAC publishes a machine-readable list, we don't need to parse the plaintext.  https://home.treasury.gov/policy-issues/financial-sanctions/specially-designated-nationals-list-data-formats-data-schemas

**Review by banshee — APPROVED** [review]

**Pull request merged**